### PR TITLE
sync with client

### PR DESCRIPTION
**Pull request by yarmolovichalex**
- sync with latest Ship React client
- add /users endpoint with paging, filtering, sorting
- update to .NET 6
- update packages

### CODE REVIEW

File: src/app/Common/DAL/BaseRepository.cs [hunk 1]
```diff
@@ -114,16 +128,35 @@ private FilterDefinition<TDocument> BuildFilterQuery(TFilter filter)
...
-            }
-
             return filterQueries.Any()
                 ? Builders<TDocument>.Filter.And(filterQueries)
                 : FilterDefinition<TDocument>.Empty;
         }
 
+        private SortDefinition<TDocument> BuildSortQuery(IList<(string Key, SortDirection Value)> sortFields = null)
+        {
+            var defaultSortFields = new List<(string, SortDirection)>
```
Comment (SergeyLukashov) [inline src/app/Common/DAL/BaseRepository.cs:138]: Do you think it's reasonable to add sort by default for all the queries?

I'm thinking about possible future performance degradations caused by this.

**Comment by SergeyLukashov** [comment]
Checked sync with new Next.js UI and found that there are two more places to fix:
- user profile, change password
- user profile, upload avatar

Please take a look.

**Comment by SergeyLukashov** [comment]
Also remembered that there we still use mongodb schemas. Not sure if it makes sense to leave them. Maybe get rid? 🤔 

Any thoughts?

**Review by SergeyLukashov — CHANGES_REQUESTED** [review]

File: src/app/Common/DAL/BaseRepository.cs (hunk 1 above)
Comment (yarmolovichalex) [reply src/app/Common/DAL/BaseRepository.cs:138]: Ok, it makes sense to remove it.

File: src/app/Common/DAL/BaseRepository.cs [hunk 2]
```diff
@@ -3,138 +3,176 @@
...
+            sortDefinitions.Add(field.Value == SortDirection.Ascending
+                ? builder.Ascending(field.Key)
+                : builder.Descending(field.Key));
         }
 
-        private void AddId(TDocument document)
+        return builder.Combine(sortDefinitions);
+    }
+
+    private static FilterDefinition<TDocument> GetFilterById(string id)
```
Comment (SergeyLukashov) [inline src/app/Common/DAL/BaseRepository.cs:161]: Looks like as an excess `static` modifier. Please remove if so.

**Review by SergeyLukashov — APPROVED** [review]

**Comment by SergeyLukashov** [comment]
Beside the note above. Please rename `DAL` folder to `Dal` too. Now only the namespace is renamed.

Once done feel free to merge 👍

**Pull request merged**